### PR TITLE
use MC instead of rAF

### PR DESCRIPTION
**Pull request by yisar**
requestAnimationFrame will be paused in when running in background tabs, Message channel will not.
This may be the best choice.

### CODE REVIEW

**Comment by jorgebucaran** [comment]
@yisar Interesting alternative!

**Comment by sergey-shpak** [comment]
@yisar 
There is no need to redraw any tab while it's in background (that is why 'requestAnimationFrame' is paused by browser), therefore you shouldn't try to redraw it with your app view. (As soon as tab becomes active 'requestAnimationFrame' is triggered and app view is rendered)

App view is only for reflecting state into dom, that's it. If you need to process something while your app is in background tab, you have to use subscriptions/effects to update state, as result re-render view. 

*Main purpose of `requestAnimationFrame` defer is to sync 'view' render with browser screen redrawing, as result getting better app performance. Btw, there is no any benefit of MC defer example comparing to simple 'setTimeout' defer.

**Comment by yisar** [comment]
> Btw, there is no any benefit of MC defer example comparing to simple 'setTimeout' defer.

It can reduce time waste as much as possible, compared with setTimeout or PostMessage.

> Main purpose of requestAnimationFrame defer is to sync 'view' render with browser screen redrawing, as result getting better app performance. 

I don't think performance is the key point? The key is that we should operate DOM less later and more safely?

**Comment by sergey-shpak** [comment]
@yisar performance is the key point, otherwise we wouldn't add 'requestAnimationFrame' defer.
> It can reduce time waste as much as possible, compared with setTimeout or PostMessage.
We don't want reduce time waste, we want to get rid of unnecessary view renders, please read my prev message once more

File: src/index.js [hunk 1]
```diff
@@ -5,10 +5,15 @@ var EMPTY_OBJ = {}
 var EMPTY_ARR = []
 var map = EMPTY_ARR.map
 var isArray = Array.isArray
-var defer =
-  typeof requestAnimationFrame !== "undefined"
-    ? requestAnimationFrame
-    : setTimeout
+var defer =(() => {
```
Comment (okwolf) [inline src/index.js]: Using an arrow function here would drop support for some of the older browsers.

Comment (yisar) [reply src/index.js]: > 在此处使用箭头功能会放弃对某些旧版浏览器的支持。

Yeap, it should be function(), but this PR seems can not merged

**Comment by jorgebucaran** [comment]
@yisar Please read @sergey-shpak's comment again https://github.com/jorgebucaran/hyperapp/pull/901#issuecomment-564982282.

**Comment by yisar** [comment]
@jorgebucaran I see, so can I closed this PR?

**Comment by sergey-shpak** [comment]
@jorgebucaran let's close this

**Comment by jorgebucaran** [comment]
Yup! Thank you @yisar @sergey-shpak! 👋

**Pull request closed without merging**